mathesar_ui/src/components/TableNameAndDescription.svelte

```diff
@@ -0,0 +1,38 @@
+<script lang="ts">
+  import type { TableEntry } from '@mathesar/api/tables';
+  import { iconTable } from '@mathesar/icons';
+  import NameWithIcon from './NameWithIcon.svelte';
+
+  export let table: Pick<TableEntry, 'name' | 'description'>;
+</script>
+
+<NameWithIcon icon={iconTable}>
+  <div class="name-and-description">
+    <span class="name" title={table.name}>{table.name}</span>
+    {#if table.description}
+      <span class="description" title={table.description}>
+        {table.description}
+      </span>
+    {/if}
+  </div>
+</NameWithIcon>
+
+<style lang="scss">
+  .name-and-description {
+    .name {
+      font-weight: bolder;
+      display: -webkit-box;
+      -webkit-line-clamp: 1;
+      -webkit-box-orient: vertical;
+      overflow: hidden;
+    }
+
+    .description {
+      display: -webkit-box;
+      -webkit-line-clamp: 1;
+      -webkit-box-orient: vertical;
+      overflow: hidden;
+      font-size: smaller;
```

Comment:
What's going on with all this vendor-prefixed CSS here? Why is this needed? I'm confused because I wouldn't expect we'd need to use any vendor prefixing here -- and if we did have vendor-prefixed CSS, I'd expect to also see the standard CSS alongside it.

Reply:
Rajat and I discussed this on a call. We're leaving the code as-is.